### DISCUSSION TITLE
Question about interactions in models

### DISCUSSION THREAD

**Discussion by jankaWIS** [bambinos/bambi#834] [Q&A]
Hi,

First of all, thank you for your excellent example gallery on your website. I have not found a place where to ask questions about it and in one of them, you write to open an issue if there's something to add or explain better. So I hope this is the right way to ask.

I was going through this tutorial about [Negative binomial regression](https://bambinos.github.io/bambi/notebooks/negative_binomial.html) where you present two models, one with and one without interaction between the two main effects. Eventually, it turns out that the interaction is not significant. But I would like to ask how it would look like if the interaction was significant and how to test for that.

And a follow-up question would be if the interaction was between two categorical variables. The example is between Program (3 levels: Academic, General, Vocational) and Math (score on a math test). But what if you had another three levels of a categorical variable, let's say a number of siblings given in three levels as 0, 1, or more than 1? How would you model such an interaction? And let's say that the effect of the number of siblings would be different between the different programs (as maybe parents can't afford to put their kids to better programs if they have more kids). How would you check that maybe there is an interaction between some of the programs and the number of siblings but not in other(s)?

Thank you in advance for your help.

**Comment by tomicapretto** [comment]
Hi @jankaWIS, thanks opening the question. I migrated it from Issue to Discussions as this is a better place for questions like this. I'll try to answer your questions.

> Eventually, it turns out that the interaction is not significant. But I would like to ask how it would look like if the interaction was significant and how to test for that.

It depends on how the interaction coefficients are implemented. In this case, we have three slopes, one for each program, and we have three parameters. But those three parameters _are not_ the slopes for the programs. They're coded like this

$$
\begin{aligned}
\beta_{1, \text{A}} &= b_1 \\
\beta_{2, \text{G}} &= b_1 +  \delta_{1, \text{G}} \\
\beta_{2, \text{V}} &= b_1 + \delta_{1, \text{V}} \\
\end{aligned}
$$

$b_1$ is `scale(math)`, $\delta_{1, \text{B}}$ is `prog:scale(math)[General]` and $\delta_{1, \text{C}}$ is `prog:scale(math)[Vocational]`

In other words, Bambi is using pivoting (or reference enconding). There's a baseline or reference slope ($b_1$) which is the slope of the reference group (Academic in this case), and the other groups have deflections around that baseline slope. 

When would be see an interaction? When any of $\beta_{1, \text{A}}$, $\beta_{2, \text{G}}$, or $\beta_{2, \text{V}}$ are different (i.e. when at least one of the slopes differ from the other). This translates to having at least one of $\delta_{1, \text{G}}$ or $\delta_{1, \text{V}}$ being different from zero. In the following plot we see they're not (because the HDIs cover 0)

![image](https://github.com/user-attachments/assets/e917eb6f-f652-4591-b4fb-c2fdf9049624)

And in this plot, comparing each curve from panel to panel, they would be different, but they're not

![image](https://github.com/user-attachments/assets/5b4e4a6b-2b11-4d54-98d3-c8ed5488f2ca)
 
>  But what if you had another three levels of a categorical variable, let's say a number of siblings given in three levels as 0, 1, or more than 1? How would you model such an interaction?

You just write it as `factor1:factor2`. It will add as many coefficients as needed.

> And let's say that the effect of the number of siblings would be different between the different programs (as maybe parents can't afford to put their kids to better programs if they have more kids). How would you check that maybe there is an interaction between some of the programs and the number of siblings but not in other(s)?

I'll try to find an example to show this

**Comment by jankaWIS** [comment]
Hi @tomicapretto,

Thank you for your answer, it's very helpful.
I assumed the encoding is done like this and looking at the tree plot would tell the interactions. But that still only looks at the 94 % HDI. Would that be enough for the claim or would you need something else, e.g., ROPE? I can't now find the notebook where I saw it, but the problem then is that usually to get ROPE, you need to either have some prior knowledge or expectations (e.g., from previous studies and literature) or somehow compute it. In which case, I would have no idea what it would be for the interaction term.
I think maybe my question here is more vague and asking what is sufficient as proof or if there are some standards or how you'd approach it.

> And in this plot, comparing each curve from panel to panel, they would be different, but they're not

If I understand you correctly, what you are proposing as proof for no interaction/interaction is to run two models, one with the term and one without and then compare the models and the results. Is that so?

> I'll try to find an example to show this

Thank you, I very much appreciate it!
